createFolders.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+DIRECTORY="$(dirname "$(readlink "$0")")"
```

Comment:
note that you do need to use the boilerplate i shared as-is to avoid some issues on _some_ POSIX systems:

```suggestion
DIRECTORY="$(readlink --canonicalize -- "$(dirname -- "$0")")"
```

the `--canonicalize` flag removes any symlinks in the path, and the `--` are needed because if some component of the path starts with `-`, the commands _may_ interpret as a flag instead of a path.

Reply:
Does `readlink` have a `canonicalize` flag ?  I didn't see that option in terminal manual file for `readlink`

Reply:
argh. does it have an `-f` flag? i was trying to avoid the shorthand names.

Reply:
![screen shot 2019-02-04 at 10 22 31 am](https://user-images.githubusercontent.com/10657206/52228280-cf3a7200-2866-11e9-9f42-ca44e7f82814.png)

Reply:
I just see the -f option for the stat command

Reply:
yuck, portable bash programming is hard.

seems like you'll have to live with some platforms being broken. or convert this into Python code.

Reply:
That was going to be my first choice in using python but I was like well let's give bash a try 😉

Reply:
i just realized there's a better alternative:

```suggestion
DIRECTORY="git rev-parse --show-toplevel
```

this should work **everywhere**.

Reply:
Works like a charm 👍